src/op_analytics/coreutils/time.py

```diff
@@ -64,3 +64,13 @@ def surrounding_dates(dateval: date) -> list[date]:
 def epoch_is_date(epoch: int) -> bool:
     """Return true if seconds since epoch is at exactly midnight."""
     return epoch % 86400 == 0
+
+
+def parse_isoformat(iso_string):
+    """Add 'Z' suffix if not present and parse ISO format strings to datetime objects."""
+    if iso_string.endswith("Z"):
+        iso_string = iso_string[:-1] + "+00:00"  # Replace 'Z' with '+00:00' for parsing
```

Comment:
still not sure why the 'Z' suffix is handled as a special case, it seems to work well with fromisoformat:
```python
In [1]: from datetime import datetime

In [2]: datetime.fromisoformat("2023-07-09T13:47:17.000Z")
Out[2]: datetime.datetime(2023, 7, 9, 13, 47, 17, tzinfo=datetime.timezone.utc)
```

I think we also need to test what we would do with an isoformatted timestamp that has a non-UTC timestamp, for example:
```
>>> datetime.fromisoformat("2024-12-10T15:30:00+02:00")
datetime.datetime(2024, 12, 10, 15, 30, tzinfo=datetime.timezone(datetime.timedelta(seconds=7200)))
```

the correct answer for the +02:00 example above should be the naive datetime object:
```
datetime(2024, 12, 10, 13, 30)
```